### PR TITLE
Add Storage Class Annotation to Jenkins Chart

### PR DESCRIPTION
**Pull request by goruha**
## what
* Add annotation `volume.beta.kubernetes.io/storage-class` -
* Added sha256 annotation to config map and secret 
* Made liveness check based on tcpSocket because with github oauth /login always returns redirect and it breaks it.

## why
* Annotation  required for support nfs provisioner and is good for support versions k8s
*  trigger automatic deployment restart

### CODE REVIEW

File: incubator/jenkins/templates/home-pvc.yaml [hunk 1]
```diff
@@ -10,7 +10,12 @@ metadata:
     release: "{{ .Release.Name }}"
     heritage: "{{ .Release.Service }}"
   annotations:
+{{- if not (empty .Values.Persistence.StorageClass) }}
+    volume.beta.kubernetes.io/storage-class:  {{ .Values.Persistence.StorageClass | quote }}
     volume.alpha.kubernetes.io/storage-class: {{ .Values.Persistence.StorageClass | quote }}
```
Comment (osterman) [inline incubator/jenkins/templates/home-pvc.yaml]: I think `volume.alpha.kubernetes.io/storage-class` has been deprecated - but cannot find where I read that

Comment (osterman) [inline incubator/jenkins/templates/home-pvc.yaml]: https://github.com/kubernetes/kubernetes.github.io/pull/2557

File: incubator/jenkins/templates/jenkins-master-deployment.yaml [hunk 2]
```diff
@@ -73,17 +75,15 @@ spec:
...
               port: {{.Values.Master.ContainerPort}}
             initialDelaySeconds: 60
             periodSeconds: 1
             timeoutSeconds: 3
             successThreshold: 1
             failureThreshold: 10
           livenessProbe:
-            httpGet:
-              path: /login
+            tcpSocket:
```
Comment (osterman) [inline incubator/jenkins/templates/jenkins-master-deployment.yaml]: Something suspect here... This has been working, why do we need to change it now? I prefer to stick with http health check. Can we ping a different URL? e.g. Image/CSS/JS asset?

File: incubator/jenkins/templates/home-pvc.yaml (hunk 1 above)
Comment (goruha) [inline incubator/jenkins/templates/home-pvc.yaml]: Done

**Pull request merged**